Source/CarthageKit/Git.swift

```diff
@@ -403,13 +403,22 @@ internal func gitmodulesEntriesInRepository(repositoryFileURL: NSURL, revision:
 		}
 }
 
+/// Returns the root directory of the given repository
+public func gitRootDirectoryForRepository(repositoryFileURL: NSURL) -> SignalProducer<String, CarthageError> {
+	let task = [ "rev-parse", "--show-toplevel" ]
+	return launchGitTask(task, repositoryFileURL: repositoryFileURL)
```

Comment:
Given that the input is already an `NSURL` I think it would make more sense that the output is too.